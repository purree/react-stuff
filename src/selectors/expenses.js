import moment from 'moment';

export default (expenses, { text, sortBy, tag, startDate, endDate }) => {
    return expenses.filter((expense) => {
        const createdAtMoment = moment(expense.createdAt);
        const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment, 'day') : true;
        const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment, 'day') : true;
        const tagMatch = tag ? expense.tag === tag : true;
        const textMatch = expense.description.toLowerCase().includes(text.toLowerCase()) || expense.note.toLowerCase().includes(text.toLowerCase());
        
        return startDateMatch && endDateMatch && textMatch && tagMatch;
    }).sort((a, b) => {
        if (sortBy === 'date') {
            return a.createdAt < b.createdAt ? 1 : -1;
        }
        else if (sortBy === 'amount') {
            return a.amount < b.amount ? 1 : -1;
        }
    });
}
