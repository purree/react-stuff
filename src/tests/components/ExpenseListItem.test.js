import React from 'react';
import { shallow } from 'enzyme';
import ExpensesListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';

test('should render ExpenseListItem', () => {
    const wrapper = shallow(<ExpensesListItem {...expenses[0]} />);
    expect(wrapper).toMatchSnapshot();
});