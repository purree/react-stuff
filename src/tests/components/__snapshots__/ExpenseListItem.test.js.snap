// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem 1`] = `
<Link
  className="list-item"
  replace={false}
  to="/edit/1"
>
  <div>
    <h3
      className="list-item__title"
    >
      Banana
    </h3>
    <span
      className="list-item__subtitle"
    >
      01 Jan 1970
    </span>
  </div>
  <h3
    className="list-item__data"
  >
    $0.90
  </h3>
</Link>
`;
